odoo/addons/base/models/res_partner.py

```diff
@@ -487,7 +487,7 @@ def onchange_company_type(self):
     @api.constrains('barcode')
     def _check_barcode_unicity(self):
         if self.barcode and self.env['res.partner'].search_count([('barcode', '=', self.barcode)]) > 1:
-            raise ValidationError('An other user already has this barcode')
+            raise ValidationError(_('An other user already has this barcode'))
```

Comment:
If we're correcting the English for other messages then we may as well fix this one too :shrug: 

```suggestion
            raise ValidationError(_('Another user already has this barcode'))
```

Reply:
@ticodoo thanks and updated